File: csharp/cars-assemble/CarsAssemble.cs
using System;

static class AssemblyLine
{
    private const int PRODUCTION_PER_HOUR = 221;
    
    private static double NormalProductionRate(int speed, double successRate) => 
        speed * PRODUCTION_PER_HOUR * successRate;

    public static double ProductionRatePerHour_(int speed) {
        if (speed >= 1 && speed <= 4) {
            return NormalProductionRate(speed, 1.0);
        } else if (speed >= 5 && speed <= 8) {
            return NormalProductionRate(speed, 0.9);
        } else if (speed == 9) {
            return NormalProductionRate(speed, 0.8);
        } else if (speed == 10) {
            return NormalProductionRate(speed, 0.77);
        } else {
            return 0;
        }
    }

    public static double ProductionRatePerHour(int speed) => speed switch {
        _ when speed == 9 => NormalProductionRate(speed, 0.8),
        _ when speed == 10 => NormalProductionRate(speed, 0.77),
        _ when speed >= 5 => NormalProductionRate(speed, 0.9),
        _ when speed >= 1 => NormalProductionRate(speed, 1.0),
        _ => 0
    };

    public static int WorkingItemsPerMinute(int speed) => 
        (int) (ProductionRatePerHour(speed) / 60);
}

File: solutions/java/cars-assemble/3/src/main/java/CarsAssemble.java
public class CarsAssemble {

    private final int PRODUCTION_PER_HOUR = 221;

    public double productionRatePerHour(int speed) {
        return speed * PRODUCTION_PER_HOUR * successRate(speed);
    }

    private double successRate(int speed) {
        if(speed == 10) return 0.77;
        if(speed == 9) return 0.8;
        if(speed >= 5) return 0.9;
        return 1;
    }

    public int workingItemsPerMinute(int speed) {
        return (int) (productionRatePerHour(speed) / 60);
    }
}